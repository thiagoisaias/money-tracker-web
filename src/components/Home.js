import React, { Fragment } from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

import LastTransactions from "./LastTransactions";
import CurrentBalance from "./CurrentBalance";

const transactions = [
  {
    id: 1,
    description: "Monthly income from freelance jobs",
    date: "05/11/2017",
    category: {
      id: 1,
      name: "Salary",
      color: "#59d9a4"
    },
    value: "1000.00",
    transactionType: "earning"
  },
  {
    id: 2,
    description: "Transference to pay mom",
    date: "08/11/2017",
    category: {
      id: 2,
      name: "Other",
      color: "#f478b8"
    },
    value: "200.00",
    transactionType: "earning"
  },
  {
    id: 3,
    description: "Uber from work to home",
    date: "30/11/2017",
    category: {
      id: 4,
      name: "Transport",
      color: "#59d4d9"
    },
    value: "16.57",
    transactionType: "expense"
  },
  {
    id: 4,
    description: "Uber from home to supermarket",
    date: "10/11/2017",
    category: {
      id: 4,
      name: "Transport",
      color: "#59d4d9"
    },
    value: "27.57",
    transactionType: "expense"
  },
  {
    id: 5,
    description: "Quintal da Varjota",
    date: "10/11/2017",
    category: {
      id: 3,
      name: "Food",
      color: "#8378f4"
    },
    value: "16.57",
    transactionType: "expense"
  },
  {
    id: 6,
    description: "Monte Carlo",
    date: "30/11/2017",
    category: {
      id: 3,
      name: "Food",
      color: "#8378f4"
    },
    value: "16.57",
    transactionType: "expense"
  }
];

const currentBalance = 2456.54;

const TransactionButton = styled.div`
  width: 150px;
  height: 50px;
  line-height: 50px;
  background-color: #8378f4;
  text-align: center;
  cursor: pointer;
  border-radius: 3px;
  color: #f2f2f2;

  &:hover {
    opacity: 0.9;
  }

  @media (max-width: 736px) {
    margin: 16px auto;
  }
`;

const Home = props => {
  return (
    <Fragment>
      <CurrentBalance currentBalance={currentBalance} />
      <NavLink to="/transaction">
        <TransactionButton> Add Transaction </TransactionButton>
      </NavLink>
      <LastTransactions transactionsList={transactions} />
    </Fragment>
  );
};

export default Home;
