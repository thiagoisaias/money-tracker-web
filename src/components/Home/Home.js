import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";
import { devices } from "../../utils/devices";

import Balance from "../Balance/Balance";
import Layout from "../Layout/Layout";
import TransactionList from "../TransactionList/TransactionList";

const TransactionButton = styled.button`
  width: 125px;
  height: 40px;
  background-color: #8378f4;
  cursor: pointer;
  border-radius: 2px;
  color: #f2f2f2;
  font-size: 12px;

  &:hover {
    opacity: 0.9;
  }

  @media ${devices.small} {
    margin: 16px auto;
  }
`;

const Home = props => {
  return (
    <Layout>
      <Balance />
      <NavLink to="/transaction">
        <TransactionButton> Add Transaction </TransactionButton>
      </NavLink>
      <TransactionList />
    </Layout>
  );
};

export default Home;
