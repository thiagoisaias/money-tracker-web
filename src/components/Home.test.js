import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Home from "./Home";

describe("Home", () => {
  const component = shallow(<Home />);

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
