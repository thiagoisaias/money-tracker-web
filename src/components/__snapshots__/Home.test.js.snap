// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Balance
          currentBalance=""
/>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/transaction"
>
          <styled.button>
                     Add Transaction 
          </styled.button>
</NavLink>,
        <TransactionList />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentBalance": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <styled.button>
             Add Transaction 
</styled.button>,
          "to": "/transaction",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Add Transaction ",
          },
          "ref": null,
          "rendered": " Add Transaction ",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Balance
            currentBalance=""
/>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/transaction"
>
            <styled.button>
                         Add Transaction 
            </styled.button>
</NavLink>,
          <TransactionList />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentBalance": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <styled.button>
               Add Transaction 
</styled.button>,
            "to": "/transaction",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Add Transaction ",
            },
            "ref": null,
            "rendered": " Add Transaction ",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
